control-plane/connect-inject/controllers/endpoints/endpoints_controller.go

```diff
@@ -482,6 +487,10 @@ func (r *Controller) createServiceRegistrations(pod corev1.Pod, serviceEndpoints
 		proxyConfig.Config[envoyPrometheusBindAddr] = prometheusScrapeListener
 	}
 
+	if r.EnableTelemetryCollector {
+		proxyConfig.Config[envoyTelemetryCollectorBindSocketDir] = "/consul/connect-inject"
```

Comment:
Not a blocker but I am thinking that maybe the directory should be `/consul/telemetry-collector` ?  Doesn't it create a <random string>.sock file?

Reply:
It does If we want to create a dedicated emptyDir volume for it we can. I figured it was a bit heavy handed given we already have a dir for connect related coordination.